flink/src/main/java/org/apache/iceberg/flink/actions/MigrateAction.java

```diff
@@ -0,0 +1,424 @@
+      Partition partition =
+          flinkHiveCatalog.getHivePartition(flinkHiveCatalog.getHiveTable(tableSource), partitionSpec);
+      inputs.add(
+          new PartitionAndLocation(partition.getSd().getLocation(), Maps.newHashMap(partitionSpec.getPartitionSpec())));
+    }
+
+    int size = partitionSpecs.size();
+    int parallelism = Math.min(size, maxParallelism);
+
+    DataStream<PartitionAndLocation> dataStream = env.fromCollection(inputs);
+    MigrateMapper migrateMapper = new MigrateMapper(spec, nameMapping, fileFormat, metricsConfig, metadataLocation);
+    DataStream<List<ManifestFile>> ds = dataStream.map(migrateMapper).setParallelism(parallelism);
+    return Lists.newArrayList(ds.executeAndCollect("migrate table :" + hiveTableName)).stream()
+        .flatMap(Collection::stream).collect(Collectors.toList());
+  }
+
+  private static class MigrateMapper extends RichMapFunction<PartitionAndLocation, List<ManifestFile>> {
+    private final PartitionSpec spec;
+    private final String nameMappingString;
+    private final FileFormat fileFormat;
+    private final MetricsConfig metricsConfig;
+    private final String metadataLocation;
+
+    MigrateMapper(PartitionSpec spec, String nameMapping, FileFormat fileFormat, MetricsConfig metricsConfig,
+                  String metadataLocation) {
+      this.spec = spec;
+      this.nameMappingString = nameMapping;
+      this.fileFormat = fileFormat;
+      this.metricsConfig = metricsConfig;
+      this.metadataLocation = metadataLocation;
+    }
+
+    @Override
+    public List<ManifestFile> map(PartitionAndLocation partitionAndLocation) {
+      Map<String, String> partitions = partitionAndLocation.getPartitionSpec();
+      String location = partitionAndLocation.getLocation();
+      Configuration conf = HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration());
+      NameMapping nameMapping = nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+      List<DataFile> files;
+      switch (fileFormat) {
+        case PARQUET:
+          files = listParquetPartition(partitions, location, spec, conf, metricsConfig);
+          break;
+
+        case ORC:
+          files = listOrcPartition(partitions, location, spec, conf, metricsConfig, nameMapping);
+          break;
+
+        default:
+          throw new UnsupportedOperationException("Unsupported file format");
+      }
+
+      return buildManifest(conf, spec, metadataLocation, files);
+    }
+
+    private List<ManifestFile> buildManifest(Configuration conf, PartitionSpec partitionSpec,
+                                             String basePath, List<DataFile> dataFiles) {
+      if (dataFiles.size() > 0) {
+        FileIO io = new HadoopFileIO(conf);
+        int subTaskId = getRuntimeContext().getIndexOfThisSubtask();
+        int attemptId = getRuntimeContext().getAttemptNumber();
+        String suffix = String.format("%d-task-%d-manifest", subTaskId, attemptId);
```

Comment:
nit: move `manifest` to the beginning? maybe sth like `manifest-task%d-attempt%d` or `manifest-%d-%d` just to be consistent?